test/index/types/types.snapshot.cc

```diff
@@ -143,3 +151,29 @@
   #undef QUALIFIED
 //       ^^^^^^^^^ reference [..] `types.cc:90:9`!
   }
+  
+  /// Restating what's already implied by the name
+  class DocumentedForwardDeclaration;
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ reference [..] DocumentedForwardDeclaration#
+  
+  class DocumentedForwardDeclaration { };
+//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition [..] DocumentedForwardDeclaration#
+//      documentation
+//      | Restating what's already implied by the name
+  
+  class Parent {};
+//      ^^^^^^ definition [..] Parent#
+  
+  class Child: Parent {};
+//      ^^^^^ definition [..] Child#
+  
+  template <class CRTPChild>
+  class CRTPBase {
+//      ^^^^^^^^ definition [..] CRTPBase#
+    void castAndDoStuff() { static_cast<CRTPChild *>(this)->doStuff(); }
```

Comment:
TODO: Maybe we should have some type information for `this` showing up in hover docs here?

Reply:
Filed https://github.com/sourcegraph/scip-clang/issues/118 -- punting this to later, since it's unrelated to this PR.